src/services/githubauth.service.js

```diff
@@ -0,0 +1,92 @@
+import GithubTokenNotFoundException from '../error/GithubTokenNotFoundException';
+import * as githubService from '../github/github.service';
+import * as encryptionUtil from '../utils/encryption';
+
+
+export default class GithubAuthenticationService {
+  constructor(githubTokenRepository, userService, logger) {
+    this.githubTokenRepository = githubTokenRepository;
+    this.userService = userService;
+    this.log = logger;
+  }
+
+  /**
+   * Checks if a user is authenticated on github. Will find the github token in
+   * githubAuthentications table. If it doesn't exist, will return false. Otherwise, it will
+   * check if the token in the table is valid on Github by making a request. If it's invalid, we
+   * will delete the github token. If it is valid, return true.
+   * @param userId the user id to check
+   * @returns {Boolean} true, if the token is valid, false otherwise.
+   */
+  async checkIfUserIsAuthenticatedOnGithub(userId) {
+    const token = await this.getGithubTokenForUser(userId);
+
+    if (token && await githubService.isUserAuthenticated(encryptionUtil.decrypt(token.token))) {
```

Comment:
Don't await in the if condition. Do it before.

Reply:
👍